moviesearch/search.py

```diff
@@ -0,0 +1,161 @@
+import json
+import re
+from BeautifulSoup import BeautifulSoup
+from mechanize import Browser
+
+
+def get_soup(movie):
+    movie = '+'.join(movie.split())
+    url = "%s%s%s" % (
+        "http://www.imdb.com/find?ref_=nv_sr_fn&q=",
+        movie,
+        "&s=all"
+    )
+    br = Browser()
+    br.open(url)
+    try:
+        link = list(br.links(url_regex=re.compile(r"/title/tt*")))[0]
+    except:
+        return ""
+    else:
+        res = br.follow_link(link)
+        soup = BeautifulSoup(res.read())
+        return soup
+
+
+def get_title(movie, soup):
+    try:
+        return soup.find('title').contents[0]
+    except:
+        return "N/A"
+
+
+def get_year(movie, soup):
+    try:
+        title_year = soup.find('span', id='titleYear')
+        year = str(title_year)
+        return re.search('.*([0-9]{4}).*', year).group(1)
+    except:
+        return "N/A"
+
+
+def get_rating(movie, soup):
+    try:
+        rate = soup.find('span', itemprop='ratingValue')
+        return str(rate.contents[0])
+    except:
+        return "N/A"
+
+
+def get_votes(movie, soup):
+    try:
+        return soup.find('span', itemprop='ratingCount').contents[0]
+    except:
+        return "N/A"
+
+
+def get_actors(movie, soup):
+    try:
+        actor_list = soup.findAll('span', itemprop='actors')
+        actors = []
+        for actor in actor_list:
+            str_act = str(actor).rpartition('itemprop="name"')[-1]
+            actors.append(re.search('\>(.*?)\<', str_act).group(1))
+        return actors
+    except:
+        return "N/A"
+
+
+def get_director(movie, soup):
+    try:
```

Comment:
Putting things in a blanket try/except block is not good. There can be so many reasons for errors and in some cases they can be handled more carefully to give better results.

You should catch specific exceptions.